### ISSUE TITLE
Different confidence score on same NLU models for the same user query.

### ISSUE THREAD

**Issue by dhruvrajkotia** [RasaHQ/rasa_nlu#1620]
**Rasa NLU version**: 0.13.8
**Operating system** (windows, osx, ...): any

**Content of model configuration file**:
```yml
pipeline:
- name: "tokenizer_whitespace"
- name: "ner_crf"
- name: "ner_synonyms"
- name: "intent_featurizer_count_vectors"
- name: "intent_classifier_tensorflow_embedding"
"epochs": 3000
```

**Issue**: When we train NLU models on same train data, and if we tried the same user query on both the models then we got different confidence score.

Is there any solution for that??

I think ideally it should be the same confidence score for both the models on the same user query.

**Comment by akelad** [comment]
Thanks for raising this issue, @MetcalfeTom  will get back to you about it soon.

**Comment by MetcalfeTom** [comment]
Hi @dhruvrajkotia,

The embeddings used for intent classification are randomly initialised, so minor perturbations are to be expected.  Are the two confidences wildly different?  I take it that this is a user query which is not in the training data

**Comment by dhruvrajkotia** [comment]
Yeah, the two confidence is widly different. It’s about 10% different all
the time you train the data.

Yeah, you are right, user query is not from training data. But it’s meaning
similar to one of the trained query.

**Comment by MetcalfeTom** [comment]
How many examples are you using to train your model?  In general we recommend the [spacy_sklearn](https://rasa.com/docs/nlu/choosing_pipeline/#the-short-answer) pipeline unless you have more than 1000 examples.

**Comment by dhruvrajkotia** [comment]
I trained model on 100 examples. and I also tried with spacy_sklearn pipeline, but it's gives too much low confidence. and I can't able to classify my intent as per user query. that's why I used tensorflow for my example. Is there any way to solve this issue??

**Comment by Ghostvv** [comment]
there is a PR that adds an ability to set random seed to an integer: https://github.com/RasaHQ/rasa_nlu/pull/1509

**Comment by MetcalfeTom** [comment]
Hi @dhruvrajkotia,

Does setting the seed solve the issue or is it more of a machine learning problem?  If you have 100 examples and try to use `tensorflow_embedding` pipeline, the model will not generalise well to newer examples.
